### ISSUE TITLE
Not purging old backups

### ISSUE THREAD

**Issue by jjroysdon** [linuxmint/timeshift#61]
**Describe the bug**
Previously, timeshift would purge old backups when the "Maximum backups exceed for backup level ..."  This no longer appears to be the case.

**To Reproduce**
1. Set hourly backups to be limit to 1
2. Run 'sudo timeshift --check' and first hourly backup will be created
3. 'sleep 1h; sudo timeshift --check' and new backup will be created but old will not be deleted.  A note will state that it should be deleted, but it won't be: 

> Maximum backups exceeded for backup level 'hourly'

**Expected behavior**
What should occur when the maximum limit is exceeded is that the oldest backups over the limit will be deleted.

**System:**
 - Linux Distribution Name and Version: Linux Mint 20.3
 - Application Version timeshift/una,now 21.09.1.1+una

**Comment by sir-ragna** [comment]
I cannot reproduce this. I tried running hourly snapshots for a whole with `while [ true ] ; do timeshift --check ; echo "sleeping ..." ; sleep 1h ; done` but snapshots do get removed.

There are cases however where snapshots are not getting removed that might take people off-guard.

For a snapshot to get removed automatically there are 3 conditions that need to be met.

- The snapshot cannot have a comment
- There must be more snapshots that the limit
- The snapshot must be older then the maximum amount of snapshot times the tag interval. This means, if you keep a maximum of 5 daily snapshots that timeshift will not remove daily snapshots if they are not older than 5 days, even if there are more than 5 snapshots present.

**Comment by jjroysdon** [comment]
> I cannot reproduce this. I tried running hourly snapshots for a whole with `while [ true ] ; do timeshift --check ; echo "sleeping ..." ; sleep 1h ; done` but snapshots do get removed.
> 
> There are cases however where snapshots are not getting removed that might take people off-guard.
> 
> For a snapshot to get removed automatically there are 3 conditions that need to be met.
> 
>     * The snapshot cannot have a comment
> 
>     * There must be more snapshots that the limit
> 
>     * The snapshot must be older then the maximum amount of snapshot times the tag interval. This means, if you keep a maximum of 5 daily snapshots that timeshift will not remove daily snapshots if they are not older than 5 days, even if there are more than 5 snapshots present.

Thanks for the reply.  I have comments on mine from time to time (pre and post kernel or driver upgrades).  I don't need them when they age out.  Would be better to have a "lock" option to keep a given snapshot from aging out (feature request), and treat comments the same as anything else.  As I have comments in many of mine, I'm betting that is the issue.

**Comment by jjroysdon** [comment]
Yup, the comments were the "problem".  After removing comments for older snapshots that exceed the limit they are now purged.  This "feature" seems broken to me.

**Comment by sir-ragna** [comment]
> Yup, the comments were the "problem". After removing comments for older snapshots that exceed the limit they are now purged. This "feature" seems broken to me.

I did bang my head against this as well. I had a hook in my package manager that created commented snapshots.

A configuration option that forces the removal of commented snapshots would be nice.

**Comment by jjroysdon** [comment]
This problem still occurs with Timeshift 24.01.1 on LM 21.3.

Any item with a comment will not be automatically purged until the comment is deleted.

**Comment by jjroysdon** [comment]
However, just tested with Timeshift 25.12.4 on LMDE 7 and the problem does not occur. Seems like it has been fixed and comments no longer hold an issue back.

I'd recommend closing this bug.

**Comment by jjroysdon** [comment]
Tested with Timeshift 25.12.4 on LMDE 7 and the problem does not occur. Seems like it has been fixed and comments no longer hold an issue back.

**Comment by jjroysdon** [comment]
Spoke too soon. Works fine with purging items with comments related to the "Boot" schedule. Isn't working with purging items with comments related to hourly.

Timeshift 25.12.4 on LMDE 7

Reports when forcing a check that the hourly amount is exceeded (duh) but won't purge the oldest as it has comments.

Remove the comment and run another check and it purges.

```
$ sudo timeshift --check
Mounted '/dev/sdb' at '/run/timeshift/2062/backup'
Boot snapshots are enabled
Last boot snapshot is 0 hours old
Hourly snapshots are enabled
Last hourly snapshot is 16 minutes old
------------------------------------------------------------------------------
**Maximum backups exceeded for backup level 'hourly'**

```

**Comment by ygerlach** [comment]
I believe that comments stop snapshots from beeing deleted is a feature. We added a tool tip to let users know here: #505 

I think we can close this issue.
